gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/CategoriesResource.java

```diff
@@ -55,20 +57,14 @@
     @Produces(MediaType.APPLICATION_JSON)
     @RequirePortalAuth
     public Response getCategories(@BeanParam PaginationParam paginationParam) {
-        // FIXME: retrieve all the apis of the user can be heavy because it involves a lot of data fetching. Find a way to just retrieve only necessary data.
-        Set<ApiEntity> apis = apiService.findPublishedByUser(getAuthenticatedUserOrNull());
+        Map<String, Long> countByCategory = apiService.countPublishedByUserGroupedByCategories(getAuthenticatedUserOrNull());
```

Comment:
Is this optimization also relevant in CategoryResource.get(categoryId) ?
Maybe we can do it the same way here ? 
WDYT ?

Reply:
:+1: